File: android/app/src/main/java/com/brekeke/phone/utils/Storage.java
package com.brekeke.phone.utils;

import com.reactnativecommunity.asyncstorage.AsyncLocalStorageUtil;
import com.reactnativecommunity.asyncstorage.ReactDatabaseSupplier;
import org.json.JSONArray;
import org.json.JSONObject;

// utils to quickly access async local storage
// see the related part in rn js for reference

public class Storage {
  // ==========================================================================
  // accounts

  private static String kAccountsAndData = "_api_profiles";
  private static String kAccounts = "profiles";
  private static String kData = "profileData";

  private static String read(String k) throws Exception {
    var ctx = Ctx.app();
    var d = ReactDatabaseSupplier.getInstance(ctx).getReadableDatabase();
    return AsyncLocalStorageUtil.getItemImpl(d, k);
  }

  private static JSONObject accountsAndData() throws Exception {
    var j = read(kAccountsAndData);
    return new JSONObject(j);
  }

  public static JSONArray accounts() throws Exception {
    return accountsAndData().optJSONArray(kAccounts);
  }

  public static JSONArray data() throws Exception {
    return accountsAndData().optJSONArray(kData);
  }

  // ==========================================================================
  // locale

  private static String kLocale = "locale";

  public static String locale() throws Exception {
    return read(kLocale);
  }
}

File: apps/brekekephone/app/android/app/src/main/java/com/brekeke/phonedev/utils/Storage.kt
package com.brekeke.phonedev.utils

import com.reactnativecommunity.asyncstorage.AsyncLocalStorageUtil
import com.reactnativecommunity.asyncstorage.ReactDatabaseSupplier
import org.json.JSONArray
import org.json.JSONObject

// utils to quickly access async local storage
// see the related part in rn js for reference

object Storage {
  private const val kAccountsAndData = "_api_profiles"
  private const val kAccounts = "profiles"
  private const val kPicker = "ringtonePicker"
  private const val kLocale = "locale"

  private fun read(k: String): String? {
    val ctx = Ctx.app()
    val d = ReactDatabaseSupplier.getInstance(ctx).readableDatabase
    return AsyncLocalStorageUtil.getItemImpl(d, k)
  }

  private fun accountsAndData(): JSONObject = JSONObject(read(kAccountsAndData)!!)

  fun accounts(): JSONArray = accountsAndData().optJSONArray(kAccounts)!!

  fun picker(): JSONObject = accountsAndData().optJSONObject(kPicker)!!

  fun locale(): String? = read(kLocale)
}
